drivers/dac/dac_dacx0508.c

```diff
@@ -0,0 +1,463 @@
+				DACX0508_MASK_GAIN_REFDIV_EN, false);
+		if (ret) {
+			LOG_ERR("GAIN Register update failed");
+			return -EIO;
+		}
+		break;
+	case DAC_REF_EXTERNAL_1_2:
+		ret = dacx0508_reg_update(dev, DACX0508_REG_CONFIG,
+				DACX0508_MASK_CONFIG_REF_PWDWN, true);
+		if (ret) {
+			LOG_ERR("GAIN Register update failed");
+			return -EIO;
+		}
+
+		ret = dacx0508_reg_update(dev, DACX0508_REG_GAIN,
+				DACX0508_MASK_GAIN_REFDIV_EN, true);
+		if (ret) {
+			LOG_ERR("GAIN Register update failed");
+			return -EIO;
+		}
+		break;
+	default:
+		LOG_ERR("unsupported channel reference type '%d'",
+			channel_cfg->reference);
+		return -ENOTSUP;
+	}
+
+	ret = dacx0508_read(dev, DACX0508_REG_STATUS, regval);
+	if (ret) {
+		LOG_ERR("Unable to read STATUS Register");
+		return -EIO;
+	}
+	if ((regval[1] & DACX0508_MASK_STATUS_REF_ALM) ==
+				DACX0508_MASK_STATUS_REF_ALM) {
+		LOG_ERR("Difference between VREF/DIV and VDD is "
+			 "below the required minimum analog threshold");
+		return -EIO;
+	}
+
+	data->configured[channel_cfg->channel_id] = true;
+
+	return 0;
+}
+
+static int dacx0508_write_value(struct device *dev, uint8_t channel,
+				uint32_t value)
+{
+	const struct dacx0508_config *config = dev->config_info;
+	struct dacx0508_data *data = dev->driver_data;
+	uint8_t regval[2];
+	int ret;
+
+	if (!data->configured[channel]) {
+		LOG_ERR("channel not initialized");
+		return -EINVAL;
+	}
+
+	if (channel > DACX0508_MAX_CHANNEL - 1) {
+		LOG_ERR("unsupported channel %d", channel);
+		return -ENOTSUP;
+	}
+
+	if (value >= (1 << config->resolution)) {
+		LOG_ERR("value %d out of range", value);
+		return -EINVAL;
+	}
+
+	regval[0] = value >> 8;
+	regval[1] = value & 0xFF;
+
+	ret = dacx0508_write(dev, DACX0508_REG_DAC0 + channel, regval);
+	if (ret) {
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static int dacx0508_soft_reset(struct device *dev)
+{
+	uint8_t regval[2] = {0, DACX0508_MASK_TRIGGER_SOFT_RESET};
+	int ret;
+
+	ret = dacx0508_write(dev, DACX0508_REG_TRIGGER, regval);
+	if (ret) {
+		return -EIO;
+	}
+	k_usleep(DACX0508_POR_DELAY);
+
+	return 0;
+}
+
+static int dacx0508_device_id_check(struct device *dev)
+{
+	const struct dacx0508_config *config = dev->config_info;
+	uint8_t regval[2] = {0, };
+	uint16_t dev_id;
+	int ret;
+
+	ret = dacx0508_read(dev, DACX0508_REG_DEVICE_ID, regval);
+	if (ret) {
+		LOG_ERR("Unable to read Device ID");
+		return -EIO;
+	}
+	dev_id = (regval[0] << 8) | regval[1];
+
+	switch (config->resolution) {
+	case 12:
+		if ((dev_id & DACX0508_MASK_DEVICE_ID_12BIT) !=
+					DACX0508_MASK_DEVICE_ID_12BIT) {
+			LOG_ERR("not match chip resolution");
+			return -EINVAL;
+		}
+		break;
+	case 14:
+		if ((dev_id & DACX0508_MASK_DEVICE_ID_14BIT) !=
+					DACX0508_MASK_DEVICE_ID_14BIT) {
+			LOG_ERR("not match chip resolution");
+			return -EINVAL;
+		}
+		break;
+	case 16:
+		if ((dev_id & DACX0508_MASK_DEVICE_ID_16BIT) != 0) {
+			LOG_ERR("not match chip resolution");
+			return -EINVAL;
+		}
+		break;
```

Comment:
Why rely on different bindings per variant when you can probe the exact chip type like this? Could we use just one binding and probe the variant instead?